Tests/Services/HealthTests/HealthTests.swift

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCHealth
+import GRPCInProcessTransport
+import XCTest
+
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+final class HealthTests: XCTestCase {
```

Comment:
We should also have tests that:

- Run check with no service name (to get the status of the overall server)
- Run watch with no service name (to get the status of the overall server)
- Run watch and update the service status to the same value multiple times, the client should only receive one update